src/pool/poolClient.ts

```diff
@@ -217,9 +219,56 @@ export class PoolEventState {
 }
 
 class UserState {
-  constructor(private contract: uma.clients.erc20.Instance) {}
-  public async read(user: string) {
+  private seen = new Set<string>();
```

Comment:
this class was updated based on copying code from the poolstate class. It now needs to handle events in a similar way as the pool state to find transfer events on the erc20